src/formatDate.js

```diff
@@ -8,7 +8,63 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObj = parseDate(date, fromFormat);
+
+  return getDate(dateObj, toFormat);
+}
+
+function parseDate(date, fromFormat) {
+  const dateObj = {};
+
+  for (let i = 0; i < 3; i++) {
+    const part = date.split(fromFormat[3])[i];
+
+    if (fromFormat[i] === 'DD') {
+      dateObj.day = part;
+    }
+
+    if (fromFormat[i] === 'MM') {
+      dateObj.month = part;
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      dateObj.year = part;
+    }
+
+    if (fromFormat[i] === 'YY') {
+      if (part < 30) {
+        dateObj.year = '20' + part;
+      } else {
+        dateObj.year = '19' + part;
```

Comment:
The condition to determine the century is not flexible for future dates. Consider parameterizing the pivot year or using a more robust method to determine the century for 'YY' to 'YYYY' conversion.